Tests/BotTests/MergeService/MergeServiceTests.swift

```diff
@@ -427,45 +404,41 @@ class MergeServiceTests: XCTestCase {
-                        sha: "abcdef",
-                        context: "",
-                        description: "N/A",
-                        state: .failure,
-                        branches: [.init(name: MergeServiceFixture.defaultBranch)]
-                    )
-                ))
+                service.sendStatusEvent(state: .failure)
 
                 scheduler.advance(by: .seconds(60))
             },
             assert: {
                 expect($0) == [
-                    makeState(status: .starting),
-                    makeState(status: .ready, pullRequests: [MergeServiceFixture.defaultTarget.reference]),
-                    makeState(status: .integrating(MergeServiceFixture.defaultTarget)),
-                    makeState(status: .runningStatusChecks(MergeServiceFixture.defaultTarget.with(mergeState: .blocked))),
-                    makeState(status: .integrationFailed(MergeServiceFixture.defaultTarget.with(mergeState: .blocked), .checksFailing)),
-                    makeState(status: .ready),
-                    makeState(status: .idle)
+                    MergeService.State.stub(status: .starting),
+                    MergeService.State.stub(status: .ready, pullRequests: [MergeServiceFixture.defaultTarget.reference]),
+                    MergeService.State.stub(status: .integrating(MergeServiceFixture.defaultTarget)),
+                    MergeService.State.stub(status: .runningStatusChecks(MergeServiceFixture.defaultTarget.with(mergeState: .blocked))),
+                    MergeService.State.stub(status: .integrationFailed(MergeServiceFixture.defaultTarget.with(mergeState: .blocked), .checksFailing)),
+                    MergeService.State.stub(status: .ready),
+                    MergeService.State.stub(status: .idle)
                 ]
             }
         )
     }
 
     func test_pull_request_with_multiple_status_checks() {
+        let requiredStatusChecks = RequiredStatusChecks.stub(indices: [0,1,2,3,4])
+
         perform(
             stubs: [
                 .getPullRequests { [MergeServiceFixture.defaultTarget.reference] },
                 .getPullRequest { _ in MergeServiceFixture.defaultTarget },
                 .postComment { _, _ in },
                 .mergeIntoBranch { _, _ in .success },
                 .getPullRequest { _ in MergeServiceFixture.defaultTarget.with(mergeState: .blocked) },
-                .getCommitStatus { _ in CommitState(state: .pending, statuses: []) },
+                .getCommitStatus { _ in CommitState.stub(states: [.pending]) },
+                .getRequiredStatusChecks{ _ in requiredStatusChecks },
                 .getPullRequest { _ in MergeServiceFixture.defaultTarget.with(mergeState: .blocked) },
-                .getCommitStatus { _ in CommitState(state: .pending, statuses: []) },
+                .getCommitStatus { _ in CommitState.stub(states: [.pending]) },
+                .getRequiredStatusChecks{ _ in requiredStatusChecks },
```

Comment:
ideally we might want to cache the result of `getRequiredStatusChecks` to avoid API calls being repeated in the future. otoh if we happen to have changed the Status Checks (e.g. we updated the CircleCI config and the job names) between the 2 calls, if we add cache we could have some strange caching issues.

I've decided to keep the decision on implementing caching for this API for a later PR (if ever)